tests/testsuite/registry_auth.rs

```diff
@@ -569,6 +569,7 @@ fn token_not_logged() {
     // 6. /api/v1/crates/new
     // 7. config.json for the "wait for publish"
     // 8. /index/3/f/foo for the "wait for publish"
-    assert_eq!(authorizations.len(), 8);
+    // FIXME: Unclear why this changed
+    assert_eq!(authorizations.len(), 10);
```

Comment:
I'm not sure why this changed. Does someone have any idea? :)

Reply:
These are the new requests:

```
main:exec: network: http-debug: * Hostname 127.0.0.1 was found in DNS cache
main:exec: network: http-debug: *   Trying 127.0.0.1:50673...
main:exec: network: http-debug: * Connected to 127.0.0.1 (127.0.0.1) port 50673
main:exec: network: http-debug: * using HTTP/1.x
main:exec: network: http-debug: > GET /index/config.json HTTP/1.1
main:exec: network: http-debug: > Host: 127.0.0.1:50673
main:exec: network: http-debug: > User-Agent: cargo/1.84.0 (b806e0a0c 2024-11-13)
main:exec: network: http-debug: > Accept-Encoding: deflate, gzip, br, zstd
main:exec: network: http-debug: > Connection: Upgrade, HTTP2-Settings
main:exec: network: http-debug: > Upgrade: h2c
main:exec: network: http-debug: > HTTP2-Settings: AAMAAABkAAQAAQAAAAIAAAAA
main:exec: network: http-debug: > cargo-protocol: version=1
main:exec: network: http-debug: > accept: text/plain
main:exec: network: http-debug: > Authorization: [REDACTED]
main:exec: network: http-debug: >
main:exec: network: http-debug: * Request completely sent off
main:exec: network: http-debug: < HTTP/1.1 200
main:exec: network: http-debug: < Content-Length: [omitted]
main:exec: network: http-debug: < Connection: close
main:exec: network: http-debug: < ETag: [omitted]
main:exec: network: http-debug: < Last-Modified: [omitted]
main:exec: network: http-debug: <
main:exec: network: http-debug: { {"dl":"http://127.0.0.1:50673/dl","api":"http://127.0.0.1:50673/","auth-required":true}
main:exec: network: http-debug: * shutting down connection #2
main:exec: network: http-debug: * Hostname 127.0.0.1 was found in DNS cache
main:exec: network: http-debug: *   Trying 127.0.0.1:50673...
main:exec: network: http-debug: * Connected to 127.0.0.1 (127.0.0.1) port 50673
main:exec: network: http-debug: * using HTTP/1.x
main:exec: network: http-debug: > GET /index/3/b/bar HTTP/1.1
main:exec: network: http-debug: > Host: 127.0.0.1:50673
main:exec: network: http-debug: > User-Agent: cargo/1.84.0 (b806e0a0c 2024-11-13)
main:exec: network: http-debug: > Accept-Encoding: deflate, gzip, br, zstd
main:exec: network: http-debug: > Connection: Upgrade, HTTP2-Settings
main:exec: network: http-debug: > Upgrade: h2c
main:exec: network: http-debug: > HTTP2-Settings: AAMAAABkAAQAAQAAAAIAAAAA
main:exec: network: http-debug: > cargo-protocol: version=1
main:exec: network: http-debug: > accept: text/plain
main:exec: network: http-debug: > if-none-match: "ddf3b656fe22a1be22b971a2c961245265908bb6a34d246d7300352152bb5565"
main:exec: network: http-debug: > Authorization: [REDACTED]
main:exec: network: http-debug: >
main:exec: network: http-debug: * Request completely sent off
main:exec: network: http-debug: < HTTP/1.1 200
main:exec: network: http-debug: < Content-Length: [omitted]
main:exec: network: http-debug: < Connection: close
main:exec: network: http-debug: < ETag: [omitted]
main:exec: network: http-debug: < Last-Modified: [omitted]
main:exec: network: http-debug: <
main:exec: network: http-debug: { {"cksum":"75120345f1869de1197d82823818e8151ad3325d58f084044181fc4457c087c4","deps":[],"features":{},"links":null,"name":"bar","vers":"1.0.0","yanked":false}
main:exec: network: http-debug: * shutting down connection #3
```

And two `GET /index/config.json` and `GET /index/3/b/bar` requests were called	from `main:exec:resolve_with_previous:resolve:block_until_ready` instead of `main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready` previously.

<details><summary>Here is the full log</summary>
<p>

### Before

```
main:exec: network: http-debug: *   Trying 127.0.0.1:50673...
main:exec: network: http-debug: * Connected to 127.0.0.1 (127.0.0.1) port 50673
main:exec: network: http-debug: * using HTTP/1.x
main:exec: network: http-debug: > GET /index/config.json HTTP/1.1
main:exec: network: http-debug: > Host: 127.0.0.1:50673
main:exec: network: http-debug: > User-Agent: cargo/1.84.0 (b806e0a0c 2024-11-13)
main:exec: network: http-debug: > Accept-Encoding: deflate, gzip, br, zstd
main:exec: network: http-debug: > Connection: Upgrade, HTTP2-Settings
main:exec: network: http-debug: > Upgrade: h2c
main:exec: network: http-debug: > HTTP2-Settings: AAMAAABkAAQAAQAAAAIAAAAA
main:exec: network: http-debug: > cargo-protocol: version=1
main:exec: network: http-debug: > accept: text/plain
main:exec: network: http-debug: >
main:exec: network: http-debug: * Request completely sent off
main:exec: network: http-debug: < HTTP/1.1 401
main:exec: network: http-debug: < Content-Length: [omitted]
main:exec: network: http-debug: < Connection: close
main:exec: network: http-debug: < WWW-Authenticate: Cargo login_url="https://test-registry-login/me"
main:exec: network: http-debug: <
main:exec: network: http-debug: { Unauthorized message from server.
main:exec: network: http-debug: * shutting down connection #0
main:exec: network: http-debug: * Hostname 127.0.0.1 was found in DNS cache
main:exec: network: http-debug: *   Trying 127.0.0.1:50673...
main:exec: network: http-debug: * Connected to 127.0.0.1 (127.0.0.1) port 50673
main:exec: network: http-debug: * using HTTP/1.x
main:exec: network: http-debug: > GET /index/config.json HTTP/1.1
main:exec: network: http-debug: > Host: 127.0.0.1:50673
main:exec: network: http-debug: > User-Agent: cargo/1.84.0 (b806e0a0c 2024-11-13)
main:exec: network: http-debug: > Accept-Encoding: deflate, gzip, br, zstd
main:exec: network: http-debug: > Connection: Upgrade, HTTP2-Settings
main:exec: network: http-debug: > Upgrade: h2c
main:exec: network: http-debug: > HTTP2-Settings: AAMAAABkAAQAAQAAAAIAAAAA
main:exec: network: http-debug: > cargo-protocol: version=1
main:exec: network: http-debug: > accept: text/plain
main:exec: network: http-debug: > Authorization: [REDACTED]
main:exec: network: http-debug: >
main:exec: network: http-debug: * Request completely sent off
main:exec: network: http-debug: < HTTP/1.1 200
main:exec: network: http-debug: < Content-Length: [omitted]
main:exec: network: http-debug: < Connection: close
main:exec: network: http-debug: < ETag: [omitted]
main:exec: network: http-debug: < Last-Modified: [omitted]
main:exec: network: http-debug: <
main:exec: network: http-debug: { {"dl":"http://127.0.0.1:50673/dl","api":"http://127.0.0.1:50673/","auth-required":true}
main:exec: network: http-debug: * shutting down connection #1
main:exec: network: http-debug: * Hostname 127.0.0.1 was found in DNS cache
main:exec: network: http-debug: *   Trying 127.0.0.1:50673...
main:exec: network: http-debug: * Connected to 127.0.0.1 (127.0.0.1) port 50673
main:exec: network: http-debug: * using HTTP/1.x
main:exec: network: http-debug: > GET /index/3/f/foo HTTP/1.1
main:exec: network: http-debug: > Host: 127.0.0.1:50673
main:exec: network: http-debug: > User-Agent: cargo/1.84.0 (b806e0a0c 2024-11-13)
main:exec: network: http-debug: > Accept-Encoding: deflate, gzip, br, zstd
main:exec: network: http-debug: > Connection: Upgrade, HTTP2-Settings
main:exec: network: http-debug: > Upgrade: h2c
main:exec: network: http-debug: > HTTP2-Settings: AAMAAABkAAQAAQAAAAIAAAAA
main:exec: network: http-debug: > cargo-protocol: version=1
main:exec: network: http-debug: > accept: text/plain
main:exec: network: http-debug: > Authorization: [REDACTED]
main:exec: network: http-debug: >
main:exec: network: http-debug: * Request completely sent off
main:exec: network: http-debug: < HTTP/1.1 404
main:exec: network: http-debug: < Content-Length: [omitted]
main:exec: network: http-debug: < Connection: close
main:exec: network: http-debug: <
main:exec: network: http-debug: { not found
main:exec: network: http-debug: * shutting down connection #2
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: *   Trying 127.0.0.1:50673...
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: * Connected to 127.0.0.1 (127.0.0.1) port 50673
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: * using HTTP/1.x
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: > GET /index/config.json HTTP/1.1
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: > Host: 127.0.0.1:50673
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: > User-Agent: cargo/1.84.0 (b806e0a0c 2024-11-13)
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: > Accept-Encoding: deflate, gzip, br, zstd
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: > Connection: Upgrade, HTTP2-Settings
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: > Upgrade: h2c
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: > HTTP2-Settings: AAMAAABkAAQAAQAAAAIAAAAA
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: > cargo-protocol: version=1
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: > accept: text/plain
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: > Authorization: [REDACTED]
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: >
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: * Request completely sent off
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: < HTTP/1.1 200
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: < Content-Length: [omitted]
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: < Connection: close
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: < ETag: [omitted]
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: < Last-Modified: [omitted]
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: <
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: { {"dl":"http://127.0.0.1:50673/dl","api":"http://127.0.0.1:50673/","auth-required":true}
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: * shutting down connection #0
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: * Hostname 127.0.0.1 was found in DNS cache
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: *   Trying 127.0.0.1:50673...
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: * Connected to 127.0.0.1 (127.0.0.1) port 50673
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: * using HTTP/1.x
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: > GET /index/3/b/bar HTTP/1.1
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: > Host: 127.0.0.1:50673
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: > User-Agent: cargo/1.84.0 (b806e0a0c 2024-11-13)
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: > Accept-Encoding: deflate, gzip, br, zstd
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: > Connection: Upgrade, HTTP2-Settings
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: > Upgrade: h2c
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: > HTTP2-Settings: AAMAAABkAAQAAQAAAAIAAAAA
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: > cargo-protocol: version=1
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: > accept: text/plain
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: > Authorization: [REDACTED]
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: >
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: * Request completely sent off
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: < HTTP/1.1 200
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: < Content-Length: [omitted]
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: < Connection: close
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: < ETag: [omitted]
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: < Last-Modified: [omitted]
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: <
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: { {"cksum":"75120345f1869de1197d82823818e8151ad3325d58f084044181fc4457c087c4","deps":[],"features":{},"links":null,"name":"bar","vers":"1.0.0","yanked":false}
main:exec:compile_ws:create_bcx:resolve_with_registry:resolve_with_previous:resolve:block_until_ready: network: http-debug: * shutting down connection #1
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: *   Trying 127.0.0.1:50673...
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: * Connected to 127.0.0.1 (127.0.0.1) port 50673
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: * using HTTP/1.x
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: > GET /dl/bar/1.0.0/download HTTP/1.1
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: > Host: 127.0.0.1:50673
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: > User-Agent: cargo/1.84.0 (b806e0a0c 2024-11-13)
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: > Accept: */*
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: > Accept-Encoding: deflate, gzip, br, zstd
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: > Connection: Upgrade, HTTP2-Settings
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: > Upgrade: h2c
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: > HTTP2-Settings: AAMAAABkAAQAAQAAAAIAAAAA
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: > Authorization: [REDACTED]
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: >
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: * Request completely sent off
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: < HTTP/1.1 200
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: < Content-Length: [omitted]
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: < Connection: close
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: <
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: { ([omitted] bytes of data)
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: * shutting down connection #0
main:exec: network: http-debug: *   Trying 127.0.0.1:50673...
main:exec: network: http-debug: * Connected to 127.0.0.1 (127.0.0.1) port 50673
main:exec: network: http-debug: * using HTTP/1.x
main:exec: network: http-debug: > PUT //api/v1/crates/new HTTP/1.1
main:exec: network: http-debug: > Host: 127.0.0.1:50673
main:exec: network: http-debug: > User-Agent: cargo/1.84.0 (b806e0a0c 2024-11-13)
main:exec: network: http-debug: > Accept-Encoding: deflate, gzip, br, zstd
main:exec: network: http-debug: > Accept: application/json
main:exec: network: http-debug: > Authorization: [REDACTED]
main:exec: network: http-debug: > Content-Length: [omitted]
main:exec: network: http-debug: >
main:exec: network: http-debug: } ([omitted] bytes of data)
main:exec: network: http-debug: } ([omitted] bytes of data)
main:exec: network: http-debug: * upload completely sent off: [omitted] bytes
main:exec: network: http-debug: < HTTP/1.1 200
main:exec: network: http-debug: < Content-Length: [omitted]
main:exec: network: http-debug: < Connection: close
main:exec: network: http-debug: <
main:exec: network: http-debug: { {"ok": true, "msg": "completed!"}
main:exec: network: http-debug: * shutting down connection #0
main:exec: network: http-debug: *   Trying 127.0.0.1:50673...
main:exec: network: http-debug: * Connected to 127.0.0.1 (127.0.0.1) port 50673
main:exec: network: http-debug: * using HTTP/1.x
main:exec: network: http-debug: > GET /index/config.json HTTP/1.1
main:exec: network: http-debug: > Host: 127.0.0.1:50673
main:exec: network: http-debug: > User-Agent: cargo/1.84.0 (b806e0a0c 2024-11-13)
main:exec: network: http-debug: > Accept-Encoding: deflate, gzip, br, zstd
main:exec: network: http-debug: > Connection: Upgrade, HTTP2-Settings
main:exec: network: http-debug: > Upgrade: h2c
main:exec: network: http-debug: > HTTP2-Settings: AAMAAABkAAQAAQAAAAIAAAAA
main:exec: network: http-debug: > cargo-protocol: version=1
main:exec: network: http-debug: > accept: text/plain
main:exec: network: http-debug: > Authorization: [REDACTED]
main:exec: network: http-debug: >
main:exec: network: http-debug: * Request completely sent off
main:exec: network: http-debug: < HTTP/1.1 200
main:exec: network: http-debug: < Content-Length: [omitted]
main:exec: network: http-debug: < Connection: close
main:exec: network: http-debug: < ETag: [omitted]
main:exec: network: http-debug: < Last-Modified: [omitted]
main:exec: network: http-debug: <
main:exec: network: http-debug: { {"dl":"http://127.0.0.1:50673/dl","api":"http://127.0.0.1:50673/","auth-required":true}
main:exec: network: http-debug: * shutting down connection #0
main:exec: network: http-debug: * Hostname 127.0.0.1 was found in DNS cache
main:exec: network: http-debug: *   Trying 127.0.0.1:50673...
main:exec: network: http-debug: * Connected to 127.0.0.1 (127.0.0.1) port 50673
main:exec: network: http-debug: * using HTTP/1.x
main:exec: network: http-debug: > GET /index/3/f/foo HTTP/1.1
main:exec: network: http-debug: > Host: 127.0.0.1:50673
main:exec: network: http-debug: > User-Agent: cargo/1.84.0 (b806e0a0c 2024-11-13)
main:exec: network: http-debug: > Accept-Encoding: deflate, gzip, br, zstd
main:exec: network: http-debug: > Connection: Upgrade, HTTP2-Settings
main:exec: network: http-debug: > Upgrade: h2c
main:exec: network: http-debug: > HTTP2-Settings: AAMAAABkAAQAAQAAAAIAAAAA
main:exec: network: http-debug: > cargo-protocol: version=1
main:exec: network: http-debug: > accept: text/plain
main:exec: network: http-debug: > Authorization: [REDACTED]
main:exec: network: http-debug: >
main:exec: network: http-debug: * Request completely sent off
main:exec: network: http-debug: < HTTP/1.1 200
main:exec: network: http-debug: < Content-Length: [omitted]
main:exec: network: http-debug: < Connection: close
main:exec: network: http-debug: < ETag: [omitted]
main:exec: network: http-debug: < Last-Modified: [omitted]
main:exec: network: http-debug: <
main:exec: network: http-debug: { {"cksum":"[omitted]","deps":[{"artifact":null,"bindep_target":null,"default_features":true,"features":[],"kind":"normal","lib":false,"name":"bar","optional":false,"package":null,"registry":null,"req":"^1.0","target":null}],"features":{},"links":null,"name":"foo","vers":"0.1.0","yanked":false}
main:exec: network: http-debug: * shutting down connection #
```

### After

```
main:exec: network: http-debug: *   Trying 127.0.0.1:50673...
main:exec: network: http-debug: * Connected to 127.0.0.1 (127.0.0.1) port 50673
main:exec: network: http-debug: * using HTTP/1.x
main:exec: network: http-debug: > GET /index/config.json HTTP/1.1
main:exec: network: http-debug: > Host: 127.0.0.1:50673
main:exec: network: http-debug: > User-Agent: cargo/1.84.0 (b806e0a0c 2024-11-13)
main:exec: network: http-debug: > Accept-Encoding: deflate, gzip, br, zstd
main:exec: network: http-debug: > Connection: Upgrade, HTTP2-Settings
main:exec: network: http-debug: > Upgrade: h2c
main:exec: network: http-debug: > HTTP2-Settings: AAMAAABkAAQAAQAAAAIAAAAA
main:exec: network: http-debug: > cargo-protocol: version=1
main:exec: network: http-debug: > accept: text/plain
main:exec: network: http-debug: >
main:exec: network: http-debug: * Request completely sent off
main:exec: network: http-debug: < HTTP/1.1 401
main:exec: network: http-debug: < Content-Length: [omitted]
main:exec: network: http-debug: < Connection: close
main:exec: network: http-debug: < WWW-Authenticate: Cargo login_url="https://test-registry-login/me"
main:exec: network: http-debug: <
main:exec: network: http-debug: { Unauthorized message from server.
main:exec: network: http-debug: * shutting down connection #0
main:exec: network: http-debug: * Hostname 127.0.0.1 was found in DNS cache
main:exec: network: http-debug: *   Trying 127.0.0.1:50673...
main:exec: network: http-debug: * Connected to 127.0.0.1 (127.0.0.1) port 50673
main:exec: network: http-debug: * using HTTP/1.x
main:exec: network: http-debug: > GET /index/config.json HTTP/1.1
main:exec: network: http-debug: > Host: 127.0.0.1:50673
main:exec: network: http-debug: > User-Agent: cargo/1.84.0 (b806e0a0c 2024-11-13)
main:exec: network: http-debug: > Accept-Encoding: deflate, gzip, br, zstd
main:exec: network: http-debug: > Connection: Upgrade, HTTP2-Settings
main:exec: network: http-debug: > Upgrade: h2c
main:exec: network: http-debug: > HTTP2-Settings: AAMAAABkAAQAAQAAAAIAAAAA
main:exec: network: http-debug: > cargo-protocol: version=1
main:exec: network: http-debug: > accept: text/plain
main:exec: network: http-debug: > Authorization: [REDACTED]
main:exec: network: http-debug: >
main:exec: network: http-debug: * Request completely sent off
main:exec: network: http-debug: < HTTP/1.1 200
main:exec: network: http-debug: < Content-Length: [omitted]
main:exec: network: http-debug: < Connection: close
main:exec: network: http-debug: < ETag: [omitted]
main:exec: network: http-debug: < Last-Modified: [omitted]
main:exec: network: http-debug: <
main:exec: network: http-debug: { {"dl":"http://127.0.0.1:50673/dl","api":"http://127.0.0.1:50673/","auth-required":true}
main:exec: network: http-debug: * shutting down connection #1
main:exec: network: http-debug: * Hostname 127.0.0.1 was found in DNS cache
main:exec: network: http-debug: *   Trying 127.0.0.1:50673...
main:exec: network: http-debug: * Connected to 127.0.0.1 (127.0.0.1) port 50673
main:exec: network: http-debug: * using HTTP/1.x
main:exec: network: http-debug: > GET /index/3/f/foo HTTP/1.1
main:exec: network: http-debug: > Host: 127.0.0.1:50673
main:exec: network: http-debug: > User-Agent: cargo/1.84.0 (b806e0a0c 2024-11-13)
main:exec: network: http-debug: > Accept-Encoding: deflate, gzip, br, zstd
main:exec: network: http-debug: > Connection: Upgrade, HTTP2-Settings
main:exec: network: http-debug: > Upgrade: h2c
main:exec: network: http-debug: > HTTP2-Settings: AAMAAABkAAQAAQAAAAIAAAAA
main:exec: network: http-debug: > cargo-protocol: version=1
main:exec: network: http-debug: > accept: text/plain
main:exec: network: http-debug: > Authorization: [REDACTED]
main:exec: network: http-debug: >
main:exec: network: http-debug: * Request completely sent off
main:exec: network: http-debug: < HTTP/1.1 404
main:exec: network: http-debug: < Content-Length: [omitted]
main:exec: network: http-debug: < Connection: close
main:exec: network: http-debug: <
main:exec: network: http-debug: { not found
main:exec: network: http-debug: * shutting down connection #2
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: *   Trying 127.0.0.1:50673...
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: * Connected to 127.0.0.1 (127.0.0.1) port 50673
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: * using HTTP/1.x
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: > GET /index/config.json HTTP/1.1
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: > Host: 127.0.0.1:50673
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: > User-Agent: cargo/1.84.0 (b806e0a0c 2024-11-13)
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: > Accept-Encoding: deflate, gzip, br, zstd
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: > Connection: Upgrade, HTTP2-Settings
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: > Upgrade: h2c
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: > HTTP2-Settings: AAMAAABkAAQAAQAAAAIAAAAA
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: > cargo-protocol: version=1
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: > accept: text/plain
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: > Authorization: [REDACTED]
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: >
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: * Request completely sent off
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: < HTTP/1.1 200
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: < Content-Length: [omitted]
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: < Connection: close
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: < ETag: [omitted]
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: < Last-Modified: [omitted]
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: <
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: { {"dl":"http://127.0.0.1:50673/dl","api":"http://127.0.0.1:50673/","auth-required":true}
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: * shutting down connection #0
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: * Hostname 127.0.0.1 was found in DNS cache
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: *   Trying 127.0.0.1:50673...
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: * Connected to 127.0.0.1 (127.0.0.1) port 50673
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: * using HTTP/1.x
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: > GET /index/3/b/bar HTTP/1.1
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: > Host: 127.0.0.1:50673
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: > User-Agent: cargo/1.84.0 (b806e0a0c 2024-11-13)
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: > Accept-Encoding: deflate, gzip, br, zstd
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: > Connection: Upgrade, HTTP2-Settings
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: > Upgrade: h2c
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: > HTTP2-Settings: AAMAAABkAAQAAQAAAAIAAAAA
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: > cargo-protocol: version=1
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: > accept: text/plain
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: > Authorization: [REDACTED]
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: >
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: * Request completely sent off
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: < HTTP/1.1 200
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: < Content-Length: [omitted]
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: < Connection: close
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: < ETag: [omitted]
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: < Last-Modified: [omitted]
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: <
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: { {"cksum":"75120345f1869de1197d82823818e8151ad3325d58f084044181fc4457c087c4","deps":[],"features":{},"links":null,"name":"bar","vers":"1.0.0","yanked":false}
main:exec:resolve_with_previous:resolve:block_until_ready: network: http-debug: * shutting down connection #1
main:exec: network: http-debug: * Hostname 127.0.0.1 was found in DNS cache
main:exec: network: http-debug: *   Trying 127.0.0.1:50673...
main:exec: network: http-debug: * Connected to 127.0.0.1 (127.0.0.1) port 50673
main:exec: network: http-debug: * using HTTP/1.x
main:exec: network: http-debug: > GET /index/config.json HTTP/1.1
main:exec: network: http-debug: > Host: 127.0.0.1:50673
main:exec: network: http-debug: > User-Agent: cargo/1.84.0 (b806e0a0c 2024-11-13)
main:exec: network: http-debug: > Accept-Encoding: deflate, gzip, br, zstd
main:exec: network: http-debug: > Connection: Upgrade, HTTP2-Settings
main:exec: network: http-debug: > Upgrade: h2c
main:exec: network: http-debug: > HTTP2-Settings: AAMAAABkAAQAAQAAAAIAAAAA
main:exec: network: http-debug: > cargo-protocol: version=1
main:exec: network: http-debug: > accept: text/plain
main:exec: network: http-debug: > Authorization: [REDACTED]
main:exec: network: http-debug: >
main:exec: network: http-debug: * Request completely sent off
main:exec: network: http-debug: < HTTP/1.1 200
main:exec: network: http-debug: < Content-Length: [omitted]
main:exec: network: http-debug: < Connection: close
main:exec: network: http-debug: < ETag: [omitted]
main:exec: network: http-debug: < Last-Modified: [omitted]
main:exec: network: http-debug: <
main:exec: network: http-debug: { {"dl":"http://127.0.0.1:50673/dl","api":"http://127.0.0.1:50673/","auth-required":true}
main:exec: network: http-debug: * shutting down connection #2
main:exec: network: http-debug: * Hostname 127.0.0.1 was found in DNS cache
main:exec: network: http-debug: *   Trying 127.0.0.1:50673...
main:exec: network: http-debug: * Connected to 127.0.0.1 (127.0.0.1) port 50673
main:exec: network: http-debug: * using HTTP/1.x
main:exec: network: http-debug: > GET /index/3/b/bar HTTP/1.1
main:exec: network: http-debug: > Host: 127.0.0.1:50673
main:exec: network: http-debug: > User-Agent: cargo/1.84.0 (b806e0a0c 2024-11-13)
main:exec: network: http-debug: > Accept-Encoding: deflate, gzip, br, zstd
main:exec: network: http-debug: > Connection: Upgrade, HTTP2-Settings
main:exec: network: http-debug: > Upgrade: h2c
main:exec: network: http-debug: > HTTP2-Settings: AAMAAABkAAQAAQAAAAIAAAAA
main:exec: network: http-debug: > cargo-protocol: version=1
main:exec: network: http-debug: > accept: text/plain
main:exec: network: http-debug: > if-none-match: "ddf3b656fe22a1be22b971a2c961245265908bb6a34d246d7300352152bb5565"
main:exec: network: http-debug: > Authorization: [REDACTED]
main:exec: network: http-debug: >
main:exec: network: http-debug: * Request completely sent off
main:exec: network: http-debug: < HTTP/1.1 200
main:exec: network: http-debug: < Content-Length: [omitted]
main:exec: network: http-debug: < Connection: close
main:exec: network: http-debug: < ETag: [omitted]
main:exec: network: http-debug: < Last-Modified: [omitted]
main:exec: network: http-debug: <
main:exec: network: http-debug: { {"cksum":"75120345f1869de1197d82823818e8151ad3325d58f084044181fc4457c087c4","deps":[],"features":{},"links":null,"name":"bar","vers":"1.0.0","yanked":false}
main:exec: network: http-debug: * shutting down connection #3
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: *   Trying 127.0.0.1:50673...
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: * Connected to 127.0.0.1 (127.0.0.1) port 50673
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: * using HTTP/1.x
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: > GET /dl/bar/1.0.0/download HTTP/1.1
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: > Host: 127.0.0.1:50673
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: > User-Agent: cargo/1.84.0 (b806e0a0c 2024-11-13)
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: > Accept: */*
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: > Accept-Encoding: deflate, gzip, br, zstd
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: > Connection: Upgrade, HTTP2-Settings
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: > Upgrade: h2c
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: > HTTP2-Settings: AAMAAABkAAQAAQAAAAIAAAAA
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: > Authorization: [REDACTED]
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: >
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: * Request completely sent off
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: < HTTP/1.1 200
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: < Content-Length: [omitted]
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: < Connection: close
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: <
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: { ([omitted] bytes of data)
main:exec:compile_ws:create_bcx:download_accessible:wait: network: http-debug: * shutting down connection #0
main:exec: network: http-debug: *   Trying 127.0.0.1:50673...
main:exec: network: http-debug: * Connected to 127.0.0.1 (127.0.0.1) port 50673
main:exec: network: http-debug: * using HTTP/1.x
main:exec: network: http-debug: > PUT //api/v1/crates/new HTTP/1.1
main:exec: network: http-debug: > Host: 127.0.0.1:50673
main:exec: network: http-debug: > User-Agent: cargo/1.84.0 (b806e0a0c 2024-11-13)
main:exec: network: http-debug: > Accept-Encoding: deflate, gzip, br, zstd
main:exec: network: http-debug: > Accept: application/json
main:exec: network: http-debug: > Authorization: [REDACTED]
main:exec: network: http-debug: > Content-Length: [omitted]
main:exec: network: http-debug: >
main:exec: network: http-debug: } ([omitted] bytes of data)
main:exec: network: http-debug: } ([omitted] bytes of data)
main:exec: network: http-debug: * upload completely sent off: [omitted] bytes
main:exec: network: http-debug: < HTTP/1.1 200
main:exec: network: http-debug: < Content-Length: [omitted]
main:exec: network: http-debug: < Connection: close
main:exec: network: http-debug: <
main:exec: network: http-debug: { {"ok": true, "msg": "completed!"}
main:exec: network: http-debug: * shutting down connection #0
main:exec: network: http-debug: *   Trying 127.0.0.1:50673...
main:exec: network: http-debug: * Connected to 127.0.0.1 (127.0.0.1) port 50673
main:exec: network: http-debug: * using HTTP/1.x
main:exec: network: http-debug: > GET /index/config.json HTTP/1.1
main:exec: network: http-debug: > Host: 127.0.0.1:50673
main:exec: network: http-debug: > User-Agent: cargo/1.84.0 (b806e0a0c 2024-11-13)
main:exec: network: http-debug: > Accept-Encoding: deflate, gzip, br, zstd
main:exec: network: http-debug: > Connection: Upgrade, HTTP2-Settings
main:exec: network: http-debug: > Upgrade: h2c
main:exec: network: http-debug: > HTTP2-Settings: AAMAAABkAAQAAQAAAAIAAAAA
main:exec: network: http-debug: > cargo-protocol: version=1
main:exec: network: http-debug: > accept: text/plain
main:exec: network: http-debug: > Authorization: [REDACTED]
main:exec: network: http-debug: >
main:exec: network: http-debug: * Request completely sent off
main:exec: network: http-debug: < HTTP/1.1 200
main:exec: network: http-debug: < Content-Length: [omitted]
main:exec: network: http-debug: < Connection: close
main:exec: network: http-debug: < ETag: [omitted]
main:exec: network: http-debug: < Last-Modified: [omitted]
main:exec: network: http-debug: <
main:exec: network: http-debug: { {"dl":"http://127.0.0.1:50673/dl","api":"http://127.0.0.1:50673/","auth-required":true}
main:exec: network: http-debug: * shutting down connection #0
main:exec: network: http-debug: * Hostname 127.0.0.1 was found in DNS cache
main:exec: network: http-debug: *   Trying 127.0.0.1:50673...
main:exec: network: http-debug: * Connected to 127.0.0.1 (127.0.0.1) port 50673
main:exec: network: http-debug: * using HTTP/1.x
main:exec: network: http-debug: > GET /index/3/f/foo HTTP/1.1
main:exec: network: http-debug: > Host: 127.0.0.1:50673
main:exec: network: http-debug: > User-Agent: cargo/1.84.0 (b806e0a0c 2024-11-13)
main:exec: network: http-debug: > Accept-Encoding: deflate, gzip, br, zstd
main:exec: network: http-debug: > Connection: Upgrade, HTTP2-Settings
main:exec: network: http-debug: > Upgrade: h2c
main:exec: network: http-debug: > HTTP2-Settings: AAMAAABkAAQAAQAAAAIAAAAA
main:exec: network: http-debug: > cargo-protocol: version=1
main:exec: network: http-debug: > accept: text/plain
main:exec: network: http-debug: > Authorization: [REDACTED]
main:exec: network: http-debug: >
main:exec: network: http-debug: * Request completely sent off
main:exec: network: http-debug: < HTTP/1.1 200
main:exec: network: http-debug: < Content-Length: [omitted]
main:exec: network: http-debug: < Connection: close
main:exec: network: http-debug: < ETag: [omitted]
main:exec: network: http-debug: < Last-Modified: [omitted]
main:exec: network: http-debug: <
main:exec: network: http-debug: { {"cksum":"[omitted]","deps":[{"artifact":null,"bindep_target":null,"default_features":true,"features":[],"kind":"normal","lib":false,"name":"bar","optional":false,"package":null,"registry":null,"req":"^1.0","target":null}],"features":{},"links":null,"name":"foo","vers":"0.1.0","yanked":false}
main:exec: network: http-debug: * shutting down connection #
```

</p>
</details> 
```

Reply:
Ah I see, thanks. So this is similar to the various changes in the tests where "Updating index" ended up in different places or one time more.

I'll update the comment above that line then.